### PR TITLE
Remove program, levels, persistent task, and submission features

### PR DESCRIPTION
**Pull request by ArmandoGraterol**
Remove deprecated features related to program, level, persistenTask, and submission from the server.

Contributes to DEV-22

### CODE REVIEW

**Review by rdig — CHANGES_REQUESTED** [review]
- Please remove the last commit. The version bump is done against the `master` branch _(I will do that myself)_
- Removing the programs broke the `UserNotifications` query. Please investigate it and fix it

See the query:
Screenshot from 2021-01-07 18-05-22

And the error:
Screenshot from 2021-01-07 18-05-36

Because of this, the user notifications don't work anymore

Screenshot from 2021-01-07 18-06-04

Please be more careful in the future removing feature, and test the branch thoroughly before submitting it for review

File: src/db/collections.ts [hunk 1]
```diff
@@ -3,11 +3,7 @@ import {
   ColonyDoc,
   DomainDoc,
   EventDoc,
-  LevelDoc,
```
Comment (rdig) [inline src/db/collections.ts:6]: The programs schemas are still needed. 

Please restore this file entirely

File: src/graphql/__tests__/apolloServer.test.ts [hunk 2]
```diff
@@ -33,15 +27,10 @@ import {
   DomainDoc,
   SuggestionDoc,
   TaskDoc,
-  TokenDoc,
```
Comment (rdig) [inline src/graphql/__tests__/apolloServer.test.ts:36]: Why was this removed? Is the `TokenDoc` not used in tests anymore?

File: src/graphql/types.ts [hunk 3]
```diff
@@ -30,7 +30,6 @@ export type Colony = {
   founder?: Maybe<User>;
   isNativeTokenExternal: Scalars['Boolean'];
   nativeTokenAddress: Scalars['String'];
-  programs: Array<Program>;
```
Comment (rdig) [inline src/graphql/types.ts:33]: Please don't touch this file manually. This is generated automatically.

So please restore this to it's original version, then run `npm run graphql:typegen` and commit the resulting file

**Review by rdig — CHANGES_REQUESTED** [review]
This is mostly good.

So files need to be restored and another one to be re-generated, but after that this will be good to go.

File: src/graphql/__tests__/apolloServer.test.ts (hunk 2 above)
Comment (ArmandoGraterol) [reply src/graphql/__tests__/apolloServer.test.ts:36]: That's right! `TokenDoc` is not used in the tests, even with the parts I've removed included

**Review by rdig — APPROVED** [review]
Seems to be all good. You can go ahead and merge it.

(Don't bother incrementing the version number now, we'll do it at some later date)
